modules/apps/foundation/address-formatter/address-formatter-api/src/main/java/com/liferay/address/formatter/util/AddressFormatterUtil.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.address.formatter.util;
+
+import com.liferay.address.formatter.AddressFormatter;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMap;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMapFactory;
+import com.liferay.portal.kernel.model.Address;
+import com.liferay.portal.kernel.model.Country;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * @author Pei-Jung Lan
+ */
+@Component(immediate = true, service = AddressFormatterUtil.class)
+public class AddressFormatterUtil {
+
+	public static String format(Address address) {
+		Country country = address.getCountry();
+
+		String a2 = country.getA2();
+
+		AddressFormatter addressFormatter = getAddressFormatter(a2);
+
+		return addressFormatter.format(address);
+	}
+
+	public static AddressFormatter getAddressFormatter(String countryA2) {
```

Comment:
Hey Jorge,

It's impossible to cover all use cases unless we're providing the formatter for all locales. It's up to each individual formatter to decide how it implements the actual display logic. The purpose of the API should provide as much utilities as possible to help devs create the formatters.

In the example you give, after the JapanFormatter detects the address is Romanized, it can choose to invoke another formatter and provide some kind of fallback mechanism to handle the case when the chosen formatter is not available, whether by calling the default USFormatter or use the Japan rules it specified.

I'll run my proposal by Sam Kong and other people that are in the email thread to get their thoughts on this.